### PR TITLE
Add planned enhancements to DG

### LINKED ISSUE #196: [PE-D][Tester A] Unlimited String Length for employees
<i><video controls><source src="https://raw.githubusercontent.com/LimJH2002/ped/main/files/e4d657ed-c46d-489e-b219-bc459260e248.mp4" type="video/mp4">Your browser does not support the video tag.</video><br>video:https://raw.githubusercontent.com/LimJH2002/ped/main/files/e4d657ed-c46d-489e-b219-bc459260e248.mp4</i>

If the name and address inputs are too long, the viewing experience become not really nice, and also it is unlikely for a person's name and address to be longer than an essay.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: LimJH2002/ped#3

### PR DESCRIPTION
**Pull request by aslam341**
Closes #196, #197, #200, #205, #208, #221.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -661,62 +661,24 @@ testers are expected to do more *exploratory* testing.
...
+We plan to add a unique ID field for each employee to allow employees with the same name and differing IDs to exist in the employee list.
+2. The current validation for emails is too lenient as it allows some invalid emails to be accepted as valid.
+For example, in the following email format `local-part@domain`, the `local-part` can exceed the maximum length of 64 characters and is considered valid in our application, when it shouldn't be.
+We plan to use a stricter regular expression to validate emails, which is mentioned [here](https://www.baeldung.com/java-email-validation-regex).
+3. Currently, there is no validation to check if the project deadline is before the task deadline.
+Therefore, users can add tasks with deadlines past the project deadline by accident without knowing.
+We plan to add a warning to the user if the user tries to add a task with a deadline past the project deadline:
+`The newly added task has a deadline past the project deadline! Check again if the details are correct and edit if needed!`
+4. Currently, a maximum of only one employee can be assigned to each task, which does not accurately reflect the real world.
```
Comment (timleow) [inline docs/DeveloperGuide.md:677]: Why did you remove @rayyan35p's explanation on using `List` to store `Employee` in a Task? Also, it might be a little difficult to read now that everything is compressed into an ordered list. It seems like there are details in the deleted code that we might still want to have.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -661,62 +661,24 @@ testers are expected to do more *exploratory* testing.
...
-
-#### Proposed Implementation
-
-In this version of TaskHub, only 1 employee can be assigned at a time to each task.
-
-Tasks in a project can be worked on by multiple people at a time, so it would be more appropriate to expand the task to take on more than 1 employee.
-
-The current implementation of tasks is already using a `List` that has been restricted to hold only one `Employee`.
-
-`List` was chosen as the data type for storing assigned `Employees` in preparation for a future iteration where more employees could be held in each task.
```
Comment (timleow) [inline docs/DeveloperGuide.md:703]: Any particular reason for removing this explanation?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -661,62 +661,24 @@ testers are expected to do more *exploratory* testing.
...
+Given below are the planned enhancements for the application.
+
+1. The current validation for duplicate employees is too strict as it does not allow employees with the same name, but different details, to exist in the employee list.
+We plan to add a unique ID field for each employee to allow employees with the same name and differing IDs to exist in the employee list.
+2. The current validation for emails is too lenient as it allows some invalid emails to be accepted as valid.
+For example, in the following email format `local-part@domain`, the `local-part` can exceed the maximum length of 64 characters and is considered valid in our application, when it shouldn't be.
+We plan to use a stricter regular expression to validate emails, which is mentioned [here](https://www.baeldung.com/java-email-validation-regex).
+3. Currently, there is no validation to check if the project deadline is before the task deadline.
+Therefore, users can add tasks with deadlines past the project deadline by accident without knowing.
+We plan to add a warning to the user if the user tries to add a task with a deadline past the project deadline:
```
Comment (timleow) [inline docs/DeveloperGuide.md:675]: Might it be better to have such code formatted as blocks rather than inline?

We plan to add a warning to the user if the user tries to add a task with a deadline past the project deadline:

```
The newly added task has a deadline past the project deadline! Check again if the details are correct and edit if needed!
```

**Review by timleow — CHANGES_REQUESTED** [review]
It seems like this section was previously better formatted and explained in detail - if your concern was to keep things concise, I believe we can still adopt some of the formatting that was in the previous code to keep things neat.

**Review by antonTan96 — APPROVED** [review]
LGTM!

**Review by timleow — APPROVED** [review]
LGTM!
